src/js/events/MetronomeClient.ts

```diff
@@ -0,0 +1,57 @@
+import { request } from "@dcos/http-service";
+import { Observable } from "rxjs/Observable";
+import "rxjs/add/operator/retryWhen";
+import "rxjs/add/operator/retry";
+
+import Config from "../config/Config";
+
+// Add interface information: https://jira.mesosphere.com/browse/DCOS-37725
+interface JobData {}
+interface JobResponse {}
+interface ScheduleData {
+  id: string
+}
+
+const defaultHeaders = {
+  'Content-Type': "application/json; charset=utf-8"
+};
+
+export function createJob(data: JobData) {
+  return request(`${Config.metronomeAPI}/v0/scheduled-jobs`, {
+    method: "POST",
+    body: JSON.stringify(data),
+    headers: defaultHeaders
+  });
+}
+
+export function fetchJobs(): Observable<JobResponse[]> {
+  return request(`${Config.metronomeAPI}/v1/jobs?embed=activeRuns&embed=schedules&embed=historySummary`, { headers: defaultHeaders });
+}
+
+export function fetchJobDetail(jobID: string): Observable<JobResponse> {
+  return request(`${Config.metronomeAPI}/v1/jobs/${jobID}?embed=activeRuns&embed=schedules&embed=historySummary`, { headers: defaultHeaders });
+}
+
+export function deleteJob(jobID: string, stopCurrentJobRuns = false): Observable<any> {
+  return request(`${Config.metronomeAPI}/v1/jobs/${jobID}?stopCurrentJobRuns=${stopCurrentJobRuns}`, { method: "DELETE", headers: defaultHeaders });
+}
+
+export function updateJob(jobID: string, data: JobData): Observable<JobResponse> {
+  return request(`${Config.metronomeAPI}/v0/scheduled-jobs/${jobID}`, {
+    method: "PUT",
+    body: JSON.stringify(data),
+    headers: defaultHeaders
+  });
+}
+
+export function runJob(jobID: string): Observable<any> {
+  return request(`${Config.metronomeAPI}/v1/jobs/${jobID}/runs`, { headers: defaultHeaders, method: "POST", body: "{}" });
+}
+
+export function stopJobRun(jobID: string, jobRunID: string): Observable<any> {
+  return request(`${Config.metronomeAPI}/v1/jobs/${jobID}/runs/${jobRunID}/actions/stop`, { headers: defaultHeaders, method: "POST", body: "{}" });
```

Comment:
shouldnt this be prettified? 🤔

Reply:
Prettier doesn't seem to work currently on dcos-ui. I created a ticket for that: https://jira.mesosphere.com/browse/DCOS_OSS-3580

Reply:
What I did for extension framework is installing proper version and running prettier manually :shrug:

Reply:
Did you run prettier eventually?